File: src/javascript/user/addressbook/list-address-books.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

const client = new FaxCoreClient();
const result = await client.post('/api/addressbook/list', {
  pagination: {
    search: process.env.FAXCORE_ADDRESS_BOOK_SEARCH || '',
    page: Number(process.env.FAXCORE_PAGE || 1),
    maxResult: Number(process.env.FAXCORE_MAX_RESULT || 25)
  }
});

printResult('Address book list', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/AddressBook/ListAddressBooks.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.AddressBook
{
    internal static class ListAddressBooks
    {
        public static Task<string> RunAsync(FaxCoreClient client)
        {
            return client.PostAsync("/api/addressbook/list", new
            {
                pagination = new
                {
                    search = Environment.GetEnvironmentVariable("FAXCORE_ADDRESS_BOOK_SEARCH") ?? string.Empty,
                    page = 1,
                    maxResult = 25
                }
            });
        }
    }
}
